packages/ui/src/components/Drawer/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { css, keyframes } from '@emotion/react'
+import styled from '@emotion/styled'
+import { type ComponentProps } from 'react'
+import { Modal, type ModalProps } from '../Modal'
+import { Separator } from '../Separator'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+import './style.css'
+
+export const SIZES = {
+  small: 356,
+  medium: 783,
+  large: 1209,
+}
+const slideIn = (translation: number) => keyframes`
+  from {
+    transform: translateX(${translation}px);
+  }
+  to {
+    transform: translateX(0);
+  }
+ `
+
+const slideAnimation = (size: 'small' | 'medium' | 'large') => {
+  if (size === 'small') return css`animation: ${slideIn(300)} linear 150ms;`
+  if (size === 'medium') return css`animation: ${slideIn(500)} linear 250ms;`
+
+  return css`animation: ${slideIn(1000)} linear 400ms;`
+}
+
+const StyledModal = styled(Modal)`
+  margin-right: 0;
+  height: 100%;
+  border-radius: 0;
+  padding: ${({ theme }) => theme.space[2]} ;
+
+  &[data-size="small"]{
+    width: ${SIZES.small}px;
+    ${slideAnimation('small')}
+  }
+
+  &[data-size="medium"]{
+    width: ${SIZES.medium}px;
+    ${slideAnimation('medium')}
+  }
+
+  &[data-size="large"]{
+    width: ${SIZES.large}px;
+    ${slideAnimation('large')}
+  }
+`
+const CustomStack = styled(Stack)`
+  height: 100%;
+`
+
+const ChildrenContainer = styled.div`
+  overflow-y: scroll;
```

Comment:
Maybe auto here instead of forcing the scroll or we will have the scroll bar that is shown all the time

![Screenshot 2024-11-25 at 10 40 59](https://github.com/user-attachments/assets/e9b3584f-7af6-46ef-999a-11c87b922a21)